superdesk/commands/clean_images.py

```diff
@@ -35,13 +36,12 @@ def run(self):
             self.__add_existing_files(used_images, upload_items)
 
             print('Number of used files: ', len(used_images))
-            current_files = superdesk.app.media.fs('upload').find({'_id': {'$nin': list(used_images)}})
 
-            for file_id in (file._id for file in current_files if str(file._id) not in used_images):
-                print('Removing unused file: ', file_id)
-                superdesk.app.media.delete(file_id)
+            if isinstance(superdesk.app.media, AmazonMediaStorage):
```

Comment:
I think it's better to move each compare function to the belonging media storage.
The call would look like: `superdesk.app.media.compare_files(used_images)`, this will remove the need for type checking here.